tests/functional/settings_secondary_emails.js

```diff
@@ -60,6 +60,49 @@ define([
       return this.remote.then(clearBrowserState());
     },
 
+    'gated in unverified session open verification same tab': function () {
+      return this.remote
+        // when an account is created, the original session is verified
+        // re-login to destroy original session and created an unverified one
+        .then(createUser(email, PASSWORD, { preVerified: true }))
+        .then(openPage(SIGNIN_URL, selectors.SIGNIN.HEADER))
+        .then(fillOutSignIn(email, PASSWORD))
+        .then(testElementExists(selectors.EMAIL.UNLOCK_BUTTON))
+
+        // unlock panel
+        .then(click(selectors.EMAIL.UNLOCK_BUTTON))
+        .then(testElementExists(selectors.EMAIL.UNLOCK_REFRESH_BUTTON))
+
+        // send and open verification in same tab
+        .then(click(selectors.EMAIL.UNLOCK_SEND_BUTTON))
+        .then(openVerificationLinkInSameTab(email, 0))
```

Comment:
One case not covered here is if I open the verification link in a new tab, the panel is not opened automatically in the new tab, I assumed that it would:

<img width="575" alt="screen shot 2017-11-08 at 14 36 39" src="https://user-images.githubusercontent.com/848085/32554656-6feec4ce-c492-11e7-90d6-9cd485b2efff.png">

Reply:
Actually, same problem when opening in the same tab. The problem is the email input is rendered but it's hidden, so `testElementExists` is an insufficient test. You need `visibleByQSA` as well.

<img width="1339" alt="screen shot 2017-11-08 at 14 42 25" src="https://user-images.githubusercontent.com/848085/32554874-1f4f0d7a-c493-11e7-92d1-75ad6c197bba.png">